### PR TITLE
added searchbar to map, added state to map

### PR DESCRIPTION
**Pull request by 0neMiss**
------------------
Trello Card(https://trello.com/c/ygKMXjbv/9-as-a-user-i-want-to-be-able-to-view-locations-on-the-map-in-real-time-while-i-look-them-up)
-------------------------
### Location:
-Map.js
-SearchBar.js
-RenderMapage.js
-LocationSubmit.js
-SetMapObject.js
### Details:
1) Map now takes data from the search bar to be passed as initial state.
2) Map has Search bar component layered on top and can update the viewport with each search.
-------------------------
## Committed Features:
1 | Write needed state management for Mapbox.
2 | Add search bar on map page.

### CODE REVIEW

**Review by dannygipson95 — DISMISSED** [review]
Everything looks good

File: src/components/common/Map.js [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
-  const [viewport, setViewport] = useState({
-    latitude: 40.73061,
-    longitude: -73.935242,
-    //The width and height has to be suffixed with vw for view width and vh for view height.
-    width: '100vw',
-    height: '100vh',
-    zoom: 8,
-  });
-  //
+  console.log(initialstate);
```
Comment (aidan-dang) [inline src/components/common/Map.js]: Shall we remove the console.log?

File: src/components/common/Map.js [hunk 2]
```diff
@@ -28,25 +36,34 @@ const Map = ({ initial_state }) => {
...
+    <div style={{ position: 'relative' }}>
+      {viewport && (
+        <ReactMapGL
+          {...viewport}
+          style={{ zIndex: 0 }}
+          mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_TOKEN}
+          mapStyle="mapbox://styles/onemiss/ckk1x7xcr16q917mjfrha2h2o"
+          onViewportChange={viewport => {
+            setViewport(viewport);
+            console.log(viewport);
```
Comment (aidan-dang) [inline src/components/common/Map.js]: Shall we remove the console.log?

File: src/utils/locationSubmit.js [hunk 3]
```diff
@@ -1,19 +1,27 @@
...
   const reqArr = value.split(', ');
   const reqBody = {
     city_name: reqArr[0],
     state_name: reqArr[1],
   };
+  let info = null;
   axios
     .post('https://h-ds2.cityspire.dev/location', reqBody)
     .then(res => {
       console.log(res.data);
```
Comment (aidan-dang) [inline src/utils/locationSubmit.js]: Shall we remove the console.log?

File: src/utils/locationSubmit.js [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
-      setter(JSON.parse(res.data)[0]);
+      //first we format the data
+      const new_value = formatter(JSON.parse(res.data)[0]);
+      //then pass it to state.
+      console.log(new_value);
```
Comment (aidan-dang) [inline src/utils/locationSubmit.js]: Shall we remove the console.log?

File: src/utils/locationSubmit.js [hunk 5]
```diff
@@ -1,19 +1,27 @@
...
+      info = new_value;
     })
     .catch(err => {
       console.error(err);
+    })
+    .finally(() => {
+      state_setter(info);
     });
 
   console.log('submitted');
```
Comment (aidan-dang) [inline src/utils/locationSubmit.js]: Shall we remove the console.log?

File: src/utils/locationSubmit.js [hunk 6]
```diff
@@ -1,19 +1,27 @@
...
+      //first we format the data
+      const new_value = formatter(JSON.parse(res.data)[0]);
+      //then pass it to state.
+      console.log(new_value);
+      info = new_value;
     })
     .catch(err => {
       console.error(err);
+    })
+    .finally(() => {
```
Comment (aidan-dang) [inline src/utils/locationSubmit.js:21]: I've learned a new thing from this. I've never known .finally() before.

**Review by aidan-dang — DISMISSED** [review]
Looks good man :)

File: src/utils/locationSubmit.js (hunk 6 above)
Comment (0neMiss) [reply src/utils/locationSubmit.js:21]: I actually am not sure if this is needed since we delt with the async issue in /renderMapPage.js

**Review by aidan-dang — APPROVED** [review]

**Pull request merged**